lib/validate_test.go

```diff
@@ -0,0 +1,87 @@
+package samplify
+
+import (
+	"testing"
+)
+
+func TestValidateSurveyURL(t *testing.T) {
+	tables := []struct {
+		id        string
+		inputURL  string
+		outputERR error
+	}{
+		{
+			"1",
+			"http://www.google.com",
+			nil,
+		},
+		{
+			"2",
+			"www.google.com",
+			nil,
+		},
+		{
+			"3",
+			"",
+			ErrURLBlank,
+		},
+		{
+			"4",
+			"ac",
+			ErrURLMinLength,
+		},
+		{
+			"5",
+			".google.com",
+			ErrURLPrefix,
+		},
+		{
+			"6",
+			"http://.google.com",
+			ErrURLHostPrefix,
+		},
+		{
+			"7",
+			"fhjksdhfjsfhjsf",
+			ErrURLHost,
+		},
+		{
+			"8",
+			"http://www.abc.com#",
+			ErrURLFragment,
+		},
+		{
+			"9",
+			"http://www.goo<gle.com",
+			ErrURLInvalid,
+		},
+		{
+			"10",
+			"http://www.google.com?a=123",
+			nil,
+		},
+		{
+			"11",
+			" ",
+			ErrURLMinLength,
+		},
+		{
+			"12",
+			"       ",
+			ErrURLHost,
+		},
+	}
+	for _, table := range tables {
+		e := ValidateSurveyURL(table.inputURL)
+		if e != table.outputERR {
```

Comment:
Can we make this better? Try to avoid long if statements and when ever possible also avoid the else statements.